### PR TITLE
Update notebooks to use CAIP 1.14 runtime

### PR DESCRIPTION
**Pull request by sararob**
@jameswex could you review? CAIP now supports TF 1.14 so I was able to simplify some of the setup steps.

Also, I figured out that if you check for the presence of `'COLAB_GPU'` in `os.environ` you can detect if a user is running in Colab. Added this to 2 of the notebooks.

### CODE REVIEW

File: xgboost_caip_e2e.ipynb [hunk 1]
```diff
@@ -19,52 +19,25 @@
...
-   "execution_count": 0,
-   "metadata": {
-    "colab": {},
-    "colab_type": "code",
-    "id": "PzmV_odNPBTb"
-   },
-   "outputs": [],
-   "source": [
-    "# You can skip this cell if you're running this from Cloud AI Platform Notebooks\n",
+    "if 'COLAB_GPU' in os.environ:\n",
```
Comment (jameswex) [inline xgboost_caip_e2e.ipynb]: is this true even when running in colab without gpu availability? i've seen another way to test for colab by i think trying to import google.colab or something, and catching any error in which case it is not colab.

Comment (jameswex) [reply xgboost_caip_e2e.ipynb]:
```
try:
  import google.colab
  IN_COLAB = True
except:
  IN_COLAB = False
```

File: keras_sklearn_compare_caip_e2e.ipynb [hunk 2]
```diff
@@ -51,9 +36,14 @@
...
    "outputs": [],
    "source": [
-    "# This is the Python package you'll use to access the What-if Tool (WIT)\n",
-    "# You can ignore the errors from this pip install\n",
-    "!pip install witwidget --quiet"
+    "# If you're running on Colab, you'll need to install the What-if Tool package and authenticate\n",
+    "import os\n",
+    "\n",
+    "if 'COLAB_GPU' in os.environ:\n",
+    "    !pip3 install witwidget --quiet\n",
```
Comment (tolga-b) [inline keras_sklearn_compare_caip_e2e.ipynb]: Is this line still correct if the user runs this colab with python2? I think pip should normally point to pip2 or pip3 depending on the active version of python, but not sure about this specific setup.

Comment (sararob) [reply keras_sklearn_compare_caip_e2e.ipynb]: Hmm, good point. If you're using a Python 3 notebook in CAIP, you need to `!pip3 install` to install packages. Any thoughts on how to handle? Should we just encourage folks to use Python 3 at the beginning of the notebook?

File: xgboost_caip_e2e.ipynb (hunk 1 above)
Comment (sararob) [reply xgboost_caip_e2e.ipynb]: That worked! Updated.

File: keras_sklearn_compare_caip_e2e.ipynb (hunk 2 above)
Comment (jameswex) [reply keras_sklearn_compare_caip_e2e.ipynb]: I think we want these notebook to work in py2 or py3 both without any need for changes from users, as our other notebooks do.

can run "!python --version" and parse the output for 2.X or 3.X and set a var and then have if statement based on it for which pip to use.

**Comment by sararob** [comment]
@jameswex @tolga-b I added a check for Python version. Could you review? Thanks!

File: keras_sklearn_compare_caip_e2e.ipynb [hunk 3]
```diff
@@ -51,9 +46,25 @@
    },
    "outputs": [],
    "source": [
-    "# This is the Python package you'll use to access the What-if Tool (WIT)\n",
-    "# You can ignore the errors from this pip install\n",
-    "!pip install witwidget --quiet"
+    "# If you're running on Colab, you'll need to install the What-if Tool package and authenticate\n",
+    "# If you're on Cloud AI Platform Notebooks, you'll need to install XGBoost on the TF instance\n",
```
Comment (jameswex) [inline keras_sklearn_compare_caip_e2e.ipynb]: only see xgboost install in third notebook. not in these first two.

File: xgboost_caip_e2e.ipynb [hunk 4]
```diff
@@ -19,52 +19,53 @@
...
-   "execution_count": 0,
+   "execution_count": 2,
    "metadata": {
     "colab": {},
     "colab_type": "code",
-    "id": "G-B8Q-4W9X6q"
+    "id": "hSFHu19Rtvbt"
    },
-   "outputs": [],
+   "outputs": [
```
Comment (jameswex) [inline xgboost_caip_e2e.ipynb]: can you clear the output of this cell before we submit the notebook?

**Pull request merged**